[Bug]: error TS2577: Return type annotation circularly references itself.

### Bug description

Steps to reproduce the problem:

1. Try to use puppeteer from Typescript.

Versions:
```
    "puppeteer": "15.3.0",
    "ts-node": "^10.8.1",
    "typescript": "4.7.4"
```
Getting error in puppeteer type definitions:
```
node_modules/puppeteer/lib/types.d.ts:3071:60 - error TS2577: Return type annotation circularly references itself.

3071         ]>>(pageFunction: Func | string, ...args: Params): Promise<Awaited<ReturnType<Func>>>;
                                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/puppeteer/lib/types.d.ts:3090:60 - error TS2577: Return type annotation circularly references itself.

3090         ]>>(pageFunction: Func | string, ...args: Params): Promise<HandleFor<Awaited<ReturnType<Func>>>>;
                                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```


### Puppeteer version

15.3.0

### Node.js version

16.14.2

### npm version

8.5.0

### What operating system are you seeing the problem on?

Linux

### Relevant log output

_No response_

--- a/.gitignore
+++ b/.gitignore
@@ -6,7 +6,7 @@ build/
 lib/
 
 # Generated files
-tsconfig.tsbuildinfo
+**/*.tsbuildinfo
 puppeteer.api.json
 puppeteer*.tgz
 yarn.lock

--- a/package.json
+++ b/package.json
@@ -62,7 +62,8 @@
   "files": [
     "lib",
     "install.js",
-    "typescript-if-required.js"
+    "typescript-if-required.js",
+    "!**/*.tsbuildinfo"
   ],
   "author": "The Chromium Authors",
   "license": "Apache-2.0",

